pkg/util/engine_version.go

```diff
@@ -0,0 +1,69 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package util
+
+import (
+	"strconv"
+	"strings"
+)
+
+// EngineVersionsMatch returns true if desired and latest engine versions match and false otherwise
+// precondition: both desiredEV and latestEV are non-nil
+// this handles the case where only the major EV is specified, e.g. "6.x" (or similar),
+// but the latest version shows the minor version, e.g. "6.0.5".
+func EngineVersionsMatch(
+	desiredEV string,
+	latestEV string,
+) bool {
+	if desiredEV == latestEV {
+		return true
+	}
+
+	dMaj, dMin, _ := versionNumbersFromString(desiredEV)
+	lMaj, lMin, _ := versionNumbersFromString(latestEV)
+	last := len(desiredEV) - 1
+
+	// if the last character of desiredEV is "x" or the major version is higher than 5, ignore patch version when comparing.
+	// See https://github.com/aws-controllers-k8s/community/issues/1737
+	if dMaj > 5 || desiredEV[last:] == "x" {
```

Comment:
qq: why 5? what if a user provides `4.x`

Reply:
This was also moved from `pkg/resource/replication_group/delta_util.go`. The reason is mentioned in this PR: https://github.com/aws-controllers-k8s/elasticache-controller/pull/115. 

> The purpose of this changes is to avoid the controller to compare patch versions when Redis version is >=6, since It's managed by AWS. More details https://github.com/aws-controllers-k8s/community/issues/1737.